src/Billing/Controllers/StripeController.cs

```diff
@@ -114,6 +117,12 @@ public async Task<IActionResult> PostWebhook([FromQuery] string key)
             return new BadRequestResult();
         }
 
+        // If the customer and server cloud regions don't match, early return 200 to avoid unnecessary errors
+        if (!await ValidateCloudRegionAsync(parsedEvent))
```

Comment:
I think we should called this immediately after below code

 ```
if (_hostingEnvironment.IsProduction() && !parsedEvent.Livemode)
        {
            _logger.LogWarning("Getting test events in production.");
            return new BadRequestResult();
        }
```
This way will avoid a lot regression testing on the webhook changes

Reply:
I'm seeing that the new validation logic is already after this bit of code! Let me know if I didn't understand your meaning